jax/_src/scipy/stats/poisson.py

```diff
@@ -114,3 +116,113 @@ def cdf(k: ArrayLike, mu: ArrayLike, loc: ArrayLike = 0) -> Array:
+    :math:`H(\mu) \approx \frac{1}{2} \log(2\pi e \mu) - \frac{1}{12\mu}`
+
+  Matches SciPy to relative error :math:`< 10^{-5}` across all regimes.
+
+  Args:
+    mu: arraylike, mean parameter of the Poisson distribution. Must be ``> 0``.
+    loc: arraylike, optional location parameter (default: 0). Shifts support to :math:`k \geq \lfloor \text{loc} \rfloor`.
+
+  Returns:
+    Array of entropy values with shape broadcast from ``mu`` and ``loc``.
+
+  Raises:
+    - ``mu <= 0`` → returns ``NaN``
+
+  Examples:
+    >>> from jax.scipy.stats import poisson
+    >>> poisson.entropy(5.0)
+    Array(2.204394, dtype=float32)
+    >>> poisson.entropy(jax.numpy.array([1, 10, 100]))
+    Array([1.3048419, 2.5614073, 3.7206903], dtype=float32)
+
+  See Also:
+    - :func:`jax.scipy.stats.poisson.pmf`
+    - :func:`jax.scipy.stats.poisson.logpmf`
+    - `SciPy docs <https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html>`_
+    
+    """
+    mu, loc = promote_args_inexact("poisson.entropy", mu, loc)
+
+    original_shape = mu.shape
+
+    mu = jnp.ravel(mu)
+    loc = jnp.ravel(loc)
+
+    # possible edge case -  mu must be positive
+    mu = jnp.where(mu > 0, mu, jnp.nan)
+
+    # choose the computation - regime based switching
+    result = jnp.where(
+        mu < 10,
+        _entropy_small_mu(mu, loc),
+        jnp.where(
+            mu < 100,
+            _entropy_medium_mu(mu, loc),
+            _entropy_large_mu(mu, loc)
+        )
+    )
+
+    # the return shape should be the original 
+    return jnp.reshape(result, original_shape)
+
+def _entropy_small_mu(mu, loc):
+    """Entropy via direct PMF summation for small μ (< 10).
+    
+    Uses adaptive upper bound k ≤ μ + 20 to capture >99.999% of mass
+    """
+    max_k = 35
+    
+    k = jnp.arange(max_k)[:, None]
+    probs = pmf(k, mu, loc)
+    
+    # Mask: only compute up to mu + 20 for each value
+    upper_bounds = jnp.ceil(mu + 20).astype(int)
+    mask = k < upper_bounds[None, :]
+    probs_masked = jnp.where(mask, probs, 0.0)
+    
+    return jnp.sum(entr(probs_masked), axis=0)
+
+def _entropy_medium_mu(mu, loc):
+    """
+    For medium mu (10-100): Adaptive bounds based on standard deviation.
+    
+    Bounds: k ≤ μ + 10√μ + 20. Caps at k=1000 (falls back to large regime).
+    """
+    max_k = 250  # Static bound for JIT. For mu<100, upper bound < 220
+    
+    k = jnp.arange(max_k)[:, None]
+    probs = pmf(k, mu, loc)
+    
+    upper_bounds = jnp.ceil(mu + 10 * jnp.sqrt(mu) + 20).astype(int)
+    mask = k < upper_bounds[None, :]
+    probs_masked = jnp.where(mask, probs, 0.0)
+    
+    return jnp.sum(entr(probs_masked), axis=0)
+
+def _entropy_large_mu(mu, loc):
+    """
+    For large mu (>= 100): Asymptotic approximation (no summation needed).
+
+    Formula: H(λ) ≈ 0.5*log(2πeλ) - 1/(12λ) + O(λ^-2)
+    """
+    return 0.5 * j_numpy.log(2 * j_numpy.pi * j_numpy.e * mu) - 1.0 / (12 * mu)
```

Comment:
You can use `jnp.log`, `np.pi`, and `np.e` here to avoid the need to import `j_numpy`.